File: app/src/main/java/co/appdev/boilerplate/data/local/SharedPrefHelper.java
package co.appdev.boilerplate.data.local;

import android.content.SharedPreferences;

import javax.inject.Inject;
import javax.inject.Singleton;

import co.appdev.boilerplate.util.Constants;

@Singleton
public class SharedPrefHelper implements UserDataHelper {

    private SharedPreferences preferences;

    @Inject
    SharedPrefHelper(SharedPreferences preferences) {
        this.preferences = preferences;
    }

    private void store(Constants.PersistenceKey key, String value) {
        preferences.edit().putString(key.toString(), value).apply();
    }

    private String retrieve(Constants.PersistenceKey key) {
        return preferences.getString(key.toString(), "");
    }

    @Override
    public void storeIsLogin(boolean is_login) {
        store(Constants.PersistenceKey.IS_LOGIN, String.valueOf(is_login));
    }

    @Override
    public boolean getIsLogin() {
        return Boolean.parseBoolean(retrieve(Constants.PersistenceKey.IS_LOGIN));
    }

    @Override
    public void clearAllPref() {
        preferences.edit().clear().apply();
    }

    public void storeToken(String token) {
        store(Constants.PersistenceKey.TOKEN, token);
    }

    public String getToken() {
        return retrieve(Constants.PersistenceKey.TOKEN);
    }

}

File: app/src/main/java/co/appdev/boilerplate/data/local/SharedPrefHelper.kt
package co.appdev.boilerplate.data.local

import android.content.SharedPreferences

import javax.inject.Inject
import javax.inject.Singleton

import co.appdev.boilerplate.util.Constants

@Singleton
class SharedPrefHelper @Inject
internal constructor(private val preferences: SharedPreferences) : UserDataHelper {

    private fun store(key: Constants.PersistenceKey, value: String) {
        preferences.edit().putString(key.toString(), value).apply()
    }

    private fun retrieve(key: Constants.PersistenceKey): String? {
        return preferences.getString(key.toString(), "")
    }

    override fun storeIsLogin(isLogin: Boolean) {
        store(Constants.PersistenceKey.IS_LOGIN, isLogin.toString())
    }

    override fun getIsLogin(): Boolean {
        return java.lang.Boolean.parseBoolean(retrieve(Constants.PersistenceKey.IS_LOGIN))
    }

    override fun clearAllPref() {
        preferences.edit().clear().apply()
    }

}